lib/galaxy_test/api/test_history_contents.py

```diff
@@ -747,3 +748,115 @@ def test_index_filter_by_type(self):
         assert len(contents_response) == expected_num_datasets
         contents_response = self._get(f"histories/{history_id}/contents?types=dataset_collection").json()
         assert len(contents_response) == expected_num_collections
+
+
+class HistoryContentsApiNearTestCase(ApiTestCase):
+    """
+    Test the /api/histories/{history_id}/contents/{direction}/{hid}/{limit} endpoint.
+    """
+    NEAR = DirectionOptions.near
+    BEFORE = DirectionOptions.before
+    AFTER = DirectionOptions.after
+
+    def setUp(self):
+        super().setUp()
+        self.dataset_populator = DatasetPopulator(self.galaxy_interactor)
+        self.dataset_collection_populator = DatasetCollectionPopulator(self.galaxy_interactor)
+
+    def _create_list_in_history(self, history_id, n=2):
+        # Creates list of size n*4 (n collections with 3 items each)
+        for _ in range(n):
+            self.dataset_collection_populator.create_list_in_history(history_id=history_id)
+
+    def _get_content(self, history_id, direction, *, hid, limit=1000):
+        return self._get(f"/api/histories/{history_id}/contents/{direction}/{hid}/{limit}").json()
+
+    def test_returned_hid_sequence_in_base_case(self):
+        with self.dataset_populator.test_history() as history_id:
+            self._create_list_in_history(history_id)
+            result = self._get_content(history_id, self.NEAR, hid=1)
+            assert len(result) == 8
+            assert result[0]['hid'] == 8
+            assert result[1]['hid'] == 7
+            assert result[2]['hid'] == 6
+            assert result[3]['hid'] == 5
+            assert result[4]['hid'] == 4
+            assert result[5]['hid'] == 3
+            assert result[6]['hid'] == 2
+            assert result[7]['hid'] == 1
+
+    def test_near_even_limit(self):
+        with self.dataset_populator.test_history() as history_id:
+            self._create_list_in_history(history_id)
+            result = self._get_content(history_id, self.NEAR, hid=5, limit=3)
+            assert len(result) == 3
+            assert result[0]['hid'] == 6  # hid + 1
+            assert result[1]['hid'] == 5  # hid
+            assert result[2]['hid'] == 4  # hid - 1
+
+    def test_near_odd_limit(self):
+        with self.dataset_populator.test_history() as history_id:
+            self._create_list_in_history(history_id)
+            result = self._get_content(history_id, self.NEAR, hid=5, limit=4)
+            assert len(result) == 4
+            assert result[0]['hid'] == 7  # hid + 1
+            assert result[1]['hid'] == 6  # hid
+            assert result[2]['hid'] == 5  # hid - 1
```

Comment:
The comments here are inaccurate, I think? The given hid was 5.

Reply:
You're right! The test is correct though. I'll adjust the comments - thanks!